### PR TITLE
feat(helm): add support for labels, annotations, podLabels and podAnnotations

### PR DESCRIPTION
**Pull request by aslafy-z**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Add ability to add labels, annotations, podAnnotations and podLabels to Helm Charts

**Which issue(s) this PR fixes**:

N/A

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
Add ability to add labels, annotations, podAnnotations and podLabels to Helm Charts
```

### CODE REVIEW

File: charts/README.md [hunk 1]
```diff
@@ -129,7 +133,12 @@ The following table lists the configurable parameters of the latest Azure File C
 | `snapshot.image.csiSnapshotController.tag`        | snapshot-controller docker image tag                       | `v5.0.1`                                                       |
 | `snapshot.image.csiSnapshotController.pullPolicy` | snapshot-controller image pull policy                      | `IfNotPresent`                                                 |
 | `snapshot.snapshotController.name`                | snapshot controller name                                   | `csi-snapshot-controller`                                                           |
-| `snapshot.snapshotController.replicas`            | the replicas of snapshot-controller                        | `2`                                                          |
+| `snapshot.snapshotController.replicas`            | the replicas of snapshot-controller                        |
+`2`
```
Comment (aslafy-z) [inline charts/README.md:137]:
```suggestion
| `snapshot.snapshotController.replicas`            | the replicas of snapshot-controller                        | `2`
```

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**